test/test_jit.py

```diff
@@ -2288,6 +2288,30 @@ def func2():
 
         self.checkScript(func2, ())
 
+    @unittest.skipIf(IS_WINDOWS, "NYI: fuser support for Windows")
+    @unittest.skipIf(not RUN_CUDA, "fuser requires CUDA")
+    @skipIfRocm
+    def test_tensor_scalar_fusion_cuda(self):
+        def should_fuse(x):
+            z = 3.
+            y = x + z
+            return x * y
+
+        def should_not_fuse(x, z):
+            y = x + int(z)
+            return x * y
```

Comment:
Why shouldn't we fuse this one?

Reply:
z isn't constant. This PR only inlines numbers into the graph fuser if they're constant.

Reply:
ohhh makes sense. Can you please just drop in a tiny comment explaining that?